src/main/java/com/mathworks/ci/TestFolders.java

```diff
@@ -0,0 +1,30 @@
+package com.mathworks.ci;
+
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import hudson.Extension;
+import hudson.model.AbstractDescribableImpl;
+import hudson.model.Descriptor;
+
+public class TestFolders extends AbstractDescribableImpl<TestFolders> {
+
+    private String testFolders;
+
+    @DataBoundConstructor
+    public TestFolders(String testFolders) {
+        this.testFolders = testFolders;
+    }
+
+    public String getTestFolders() {
+        return this.testFolders;
+    }
+
+    @Extension
+    public static final class DescriptorImpl extends Descriptor<TestFolders> {
+
+        @Override
+        public String getDisplayName() {
```

Comment:
Is returning empty string the right thing to do? If the aim is not display it, there may be other ways to do this. 

https://javadoc.jenkins-ci.org/hudson/model/Descriptor.html#getDisplayName--

Reply:
good point actually overriding this method is not mandatory at all unless we want it to be shown on UI somehow.  We could fall back to default behaviour by not overriding it.  Done it for all such descriptor which returned empty string.